projects/aca-shared/rules/src/app.rules.spec.ts

```diff
@@ -1003,6 +967,978 @@ describe('app.evaluators', () => {
+      context.navigation.url = '/personal-files/test';
+      context.selection.file = null;
+      expect(app.canViewFile(context)).toBeFalse();
+    });
+
+    it('should return true when user is not in trashcan and file is selected', () => {
+      context.navigation.url = '/personal-files/test';
+      context.selection.file = {} as any;
+      expect(app.canViewFile(context)).toBeTrue();
+    });
+  });
+
+  describe('canLeaveLibrary', () => {
+    const context = createTestContext();
+
+    it('should return false when no library is selected', () => {
+      context.selection.library = null;
+      expect(app.canLeaveLibrary(context)).toBeFalse();
+    });
+
+    it('should return false when user does not have library role', () => {
+      context.selection.library = { entry: { role: null } } as any;
+      expect(app.canLeaveLibrary(context)).toBeFalse();
+    });
+
+    it('should return true when user has a library role', () => {
+      context.selection.library = { entry: { role: 'test' } } as any;
+      expect(app.canLeaveLibrary(context)).toBeTrue();
+    });
+  });
+
+  describe('canToggleSharedLink', () => {
+    const context = createTestContext();
+    context.repository.status = new StatusInfo();
+
+    it('should return false when no file is selected', () => {
+      context.selection.file = null;
+      expect(app.canToggleSharedLink(context)).toBeFalse();
+    });
+
+    it('should return false when file is not shared and user cannot share it', () => {
+      context.selection.file = { entry: { properties: {} } } as any;
+      context.repository.status.isQuickShareEnabled = false;
+      expect(app.canToggleSharedLink(context)).toBeFalse();
+    });
+
+    it('should return true when file is already shared', () => {
+      context.selection.file = { entry: { properties: { 'qshare:sharedId': 'some-id' } } } as any;
+      context.repository.status.isQuickShareEnabled = false;
+      expect(app.canToggleSharedLink(context)).toBeTrue();
+    });
+
+    it('should return true when file can be shared', () => {
+      context.selection.file = { entry: { properties: {} } } as any;
+      context.repository.status.isQuickShareEnabled = true;
+      expect(app.canToggleSharedLink(context)).toBeTrue();
+    });
+  });
+
+  describe('canShowInfoDrawer', () => {
+    const context = createTestContext();
+
+    it('should return false when nothing is selected', () => {
+      context.selection.isEmpty = true;
+      expect(app.canShowInfoDrawer(context)).toBeFalse();
+    });
+
+    it('should return false when user is in libraries or trashcan', () => {
+      context.selection.isEmpty = false;
+      context.navigation.url = '/trashcan/test';
+      expect(app.canShowInfoDrawer(context)).toBeFalse();
+
+      context.navigation.url = '/test/libraries';
+      expect(app.canShowInfoDrawer(context)).toBeFalse();
+    });
+
+    it('should return true when selection exists and user is not it trashcan or libraries', () => {
```

Comment:
Typo, should be 'user is not **in** trashcan or libraries'

Reply:
Fixed